### PR TITLE
Organisation pull request

### PR DESCRIPTION
**Pull request by Grey-rgb**
Model:
- Organisation
- OrganisationValidator

Exceptions:
- OrganisationNameEmptyException
- OrganisationNameTooLongException
- OrganisationNameTooShortException
- OrganisationDocumentationAlreadyExistsException
- OrganisationDocumentationNotFoundException
- OrganisationNeedsAnOwnerException
- OrganisationOwnerAlreadyExistsException
- OrganisationOwnerNotFoundException

Tests:
- OrganisationTests

### CODE REVIEW

File: src/core/domain/exceptions/models/Organisation/OrganisationDocumentationAlreadyExistsException.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    /// <param name="message">Customized message.</param>
+    /// <param name="innerException">Inner exception.</param>
+    public OrganisationDocumentationAlreadyExistsException(string message, Exception innerException) : base(message, innerException) { }
+
+    /// <summary>
+    /// Used for serialization.
+    /// </summary>
+    /// <param name="info"></param>
+    /// <param name="context"></param>
+    protected OrganisationDocumentationAlreadyExistsException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationDocumentationAlreadyExistsException.cs:31]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationDocumentationAlreadyExistsException.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+using System.Runtime.Serialization;
+
+namespace domain.exceptions.models.organisation;
+
+[Serializable]
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationDocumentationAlreadyExistsException.cs:5]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationDocumentationAlreadyExistsException.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationDocumentationAlreadyExistsException.cs:1]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationDocumentationNotFoundException.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
+using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationDocumentationNotFoundException.cs:1]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationDocumentationNotFoundException.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+namespace domain.exceptions.orgamodels.organisationnisation;
+
+[Serializable]
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationDocumentationNotFoundException.cs:5]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationDocumentationNotFoundException.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    /// <param name="message">Customized message.</param>
+    /// <param name="innerException">Inner exception.</param>
+    public OrganisationDocumentationNotFoundException(string message, Exception innerException) : base(message, innerException) { }
+
+    /// <summary>
+    /// Used for serialization.
+    /// </summary>
+    /// <param name="info"></param>
+    /// <param name="context"></param>
+    protected OrganisationDocumentationNotFoundException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationDocumentationNotFoundException.cs:31]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameEmptyException.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
+using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameEmptyException.cs:1]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameEmptyException.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace domain.exceptions.models.organisation;
+
+/// <summary>
+/// Exception for when an Organisation is created without a title.
+/// </summary>
+[Serializable]
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameEmptyException.cs:8]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameEmptyException.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    /// <param name="message">Customized message.</param>
+    /// <param name="innerException">Inner exception.</param>
+    public OrganisationNameEmptyException(string message, Exception innerException) : base(message, innerException) { }
+    
+    /// <summary>
+    /// Used for serialization.
+    /// </summary>
+    /// <param name="info"></param>
+    /// <param name="context"></param>
+    protected OrganisationNameEmptyException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameEmptyException.cs:34]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooLongException.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
+using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooLongException.cs:1]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooLongException.cs [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace domain.exceptions.models.organisation;
+
+[Serializable]
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooLongException.cs:5]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooLongException.cs [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+public class OrganisationNameTooLongException : Exception
+{
+    public OrganisationNameTooLongException() : base("Title is too long, it cannot be more than 100 characters.") { }
+    
+    public OrganisationNameTooLongException(string message) : base(message) { }
+    
+    public OrganisationNameTooLongException(string message, Exception innerException) : base(message, innerException) { }
+    
+    protected OrganisationNameTooLongException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooLongException.cs:14]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooLongException.cs [hunk 13]
```diff
@@ -0,0 +1,15 @@
+using System.Runtime.Serialization;
+
+namespace domain.exceptions.models.organisation;
+
+[Serializable]
+public class OrganisationNameTooLongException : Exception
+{
+    public OrganisationNameTooLongException() : base("Title is too long, it cannot be more than 100 characters.") { }
+    
+    public OrganisationNameTooLongException(string message) : base(message) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooLongException.cs:10]: Add documentation

File: src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooShortException.cs [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+
+public class OrganisationNameTooShortException : Exception
+{
+    public OrganisationNameTooShortException() : base("Title is too short, it cannot be less than 2 characters.") { }
+    
+    public OrganisationNameTooShortException(string message) : base(message) { }
+    
+    public OrganisationNameTooShortException(string message, Exception innerException) : base(message, innerException) { }
+    
+    protected OrganisationNameTooShortException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooShortException.cs:13]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooShortException.cs [hunk 15]
```diff
@@ -0,0 +1,14 @@
+using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooShortException.cs:1]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooShortException.cs [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace domain.exceptions.models.organisation;
+
+public class OrganisationNameTooShortException : Exception
+{
+    public OrganisationNameTooShortException() : base("Title is too short, it cannot be less than 2 characters.") { }
+    
+    public OrganisationNameTooShortException(string message) : base(message) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationName/OrganisationNameTooShortException.cs:9]: Add documentation

File: src/core/domain/exceptions/models/Organisation/OrganisationNeedsAnOwnerException.cs [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+    /// <param name="message">Customized message.</param>
+    /// <param name="innerException">Inner exception.</param>
+    public OrganisationNeedsAnOwnerException(string message, Exception innerException) : base(message, innerException) { }
+    
+    /// <summary>
+    /// Used for serialization.
+    /// </summary>
+    /// <param name="info"></param>
+    /// <param name="context"></param>
+    protected OrganisationNeedsAnOwnerException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationNeedsAnOwnerException.cs:31]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationNeedsAnOwnerException.cs [hunk 18]
```diff
@@ -0,0 +1,32 @@
+using System.Runtime.Serialization;
+
+namespace domain.exceptions.models.organisation;
+
+[Serializable]
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationNeedsAnOwnerException.cs:5]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationNeedsAnOwnerException.cs [hunk 19]
```diff
@@ -0,0 +1,32 @@
+using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationNeedsAnOwnerException.cs:1]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationOwnerNotFoundException.cs [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+    /// <param name="message">Customized message.</param>
+    /// <param name="innerException">Inner exception.</param>
+    public OrganisationOwnerNotFoundException(string message, Exception innerException) : base(message, innerException) { }
+    
+    /// <summary>
+    /// Used for serialization.
+    /// </summary>
+    /// <param name="info"></param>
+    /// <param name="context"></param>
+    protected OrganisationOwnerNotFoundException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationOwnerNotFoundException.cs:31]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationOwnerNotFoundException.cs [hunk 21]
```diff
@@ -0,0 +1,32 @@
+using System.Runtime.Serialization;
+
+namespace domain.exceptions.models.organisation;
+
+[Serializable]
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationOwnerNotFoundException.cs:5]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationOwnerNotFoundException.cs [hunk 22]
```diff
@@ -0,0 +1,32 @@
+using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationOwnerNotFoundException.cs:1]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationOwnerAlreadyExistsException.cs [hunk 23]
```diff
@@ -0,0 +1,32 @@
+using System.Runtime.Serialization;
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationOwnerAlreadyExistsException.cs:1]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationOwnerAlreadyExistsException.cs [hunk 24]
```diff
@@ -0,0 +1,32 @@
...
+
+namespace domain.exceptions.models.organisation;
+
+[Serializable]
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationOwnerAlreadyExistsException.cs:5]: Remove.

File: src/core/domain/exceptions/models/Organisation/OrganisationOwnerAlreadyExistsException.cs [hunk 25]
```diff
@@ -0,0 +1,32 @@
...
+    /// <param name="message">Customized message.</param>
+    /// <param name="innerException">Inner exception.</param>
+    public OrganisationOwnerAlreadyExistsException(string message, Exception innerException) : base(message, innerException) { }
+
+    /// <summary>
+    /// Used for serialization.
+    /// </summary>
+    /// <param name="info"></param>
+    /// <param name="context"></param>
+    protected OrganisationOwnerAlreadyExistsException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (MalasataXD) [inline src/core/domain/exceptions/models/Organisation/OrganisationOwnerAlreadyExistsException.cs:31]: Remove.

**Review by MalasataXD — CHANGES_REQUESTED** [review]
Minor changes. But you got it! 😉

File: src/Tests/UnitTests/models/Organisation/OrganisationTests.cs [hunk 26]
```diff
@@ -6,7 +6,7 @@
 
 namespace UnitTests.models.organisation
 {
-    public class OrganisationModelTests
+    public class OrganisationTests
```
Comment (MalasataXD) [inline src/Tests/UnitTests/models/Organisation/OrganisationTests.cs:9]: Just a minor thing.. It should actually be called `OrganisationModelTests` :wink

**Review by MalasataXD — CHANGES_REQUESTED** [review]
If you just fix my one little comment. It should be ready to merge 👍

**Review by MalasataXD — APPROVED** [review]
LGTM 👍

**Review by MalasataXD — COMMENTED** [review]
LGTM 👍

**Review by MalasataXD — APPROVED** [review]
LGTM 🥇

**Review by MalasataXD — APPROVED** [review]
LETS GOO PLEASE :3

**Review by BjarkeBP — APPROVED** [review]
LGTM

**Pull request merged**